lib/client-api/src/args.ts

```diff
@@ -50,3 +53,42 @@ export const combineArgs = (value: any, update: any): Args => {
     return acc;
   }, {} as any);
 };
+
+export const validateOptions = (args: Args, argTypes: ArgTypes): Args => {
+  return Object.entries(argTypes).reduce((acc, [key, { options }]) => {
+    if (!options) {
+      acc[key] = args[key];
+      return acc;
+    }
+
+    if (!Array.isArray(options)) {
+      once.warn(dedent`
+        Invalid argType: '${key}.options' should be an array.
+
+        More info: https://storybook.js.org/docs/react/api/argtypes
+      `);
+      acc[key] = args[key];
+      return acc;
+    }
+
+    if (options.some((opt) => opt && ['object', 'function'].includes(typeof opt))) {
+      once.warn(dedent`
```

Comment:
Nice, I like this pattern better than `util-deprecate` 💯

Reply:
Are we using this `once.warn` for any other purposes? Would be nice to be able to globally search for something with every major release and get a list of all the stuff we have to strip out of the code.

Reply:
Look for the word "deprecate" ?